src/utils/DownloadManager.js

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+const { inspect } = require('util');
+const URI = require('uri-js');
+const { setdefault } = require('ferrum');
+const { fetch, TimeoutError } = require('@adobe/helix-fetch');
+
+class DownloadManager {
+  constructor(context, action) {
+    this._context = context;
+    this._action = action;
+    this._queue = [];
+
+    const { logger, secrets } = this._action;
+    if (!secrets || !secrets.HTTP_TIMEOUT) {
+      logger.warn('No HTTP timeout set, risk of denial-of-service');
+    }
+  }
+
+  get tasks() {
+    return this._queue;
+  }
+
+  get githubRootPath() {
+    return this._action.secrets.REPO_RAW_ROOT;
+  }
+
+  get timeout() {
+    return this._action.secrets.HTTP_TIMEOUT;
+  }
+
+  get githubToken() {
+    const { request = {}, secrets } = this._action;
+    return (secrets && secrets.GITHUB_TOKEN) || (request.headers ? request.headers['x-github-token'] : '');
+  }
+
+  computeGithubURI(owner, repo, ref, path) {
+    const rootURI = URI.parse(this.githubRootPath);
+    const rootPath = rootURI.path;
+    // remove double slashes
+    const fullPath = `${rootPath}/${owner}/${repo}/${ref}/${path}`.replace(
+      /\/+/g,
+      '/',
+    );
+
+    return URI.serialize({
+      scheme: rootURI.scheme,
+      host: rootURI.host,
+      port: rootURI.port,
+      path: fullPath,
+    });
+  }
+
+  /**
+   * Schedules a task that fetches a web resource.
+   * @param {object} opts options.
+   * @param {object} opts.uri URI to download
+   * @param {object} opts.options Fetch options.
+   * @param {string} opts.id Some id to later identify the task.
+   * @param {boolean} opts.errorOn404 Treat 404 as error.
+   * @return {Promise} Promise that resolves with the fetch result.
+   */
+
+  fetch(opts) {
+    const { request = {}, logger } = this._action;
+    const context = this._context;
+
+    const { uri, options = {} } = opts;
+    if (!uri) {
+      throw new Error('Unknown uri, cannot fetch content');
+    }
+
+    if (!('timeout' in options)) {
+      options.timeout = this.timeout;
+    }
+
+    if (!('cache' in options)) {
+      options.cache = request.headers ? request.headers['Cache-Control'] : '';
+    }
+    options.cache = '';
```

Comment:
@stefan-guggisberg when I comment this line, the `testHTML.js` tests don't terminate....probably related to helix-fetch.
also: is there a way to easily disable the cache? I think it confuses pollyjs.

Reply:
`options.cache = 'no-store';` disables the cache.